cms/djangoapps/contentstore/tests/test_orphan.py

```diff
@@ -128,3 +131,88 @@ def test_not_permitted(self, default_store):
+
+        # Verify `OrphanVert` is an orphan
+        self.assertIn(orphan_vertical.location, self.store.get_orphans(course.id))
+
+        # Verify `multi_parent_html` is child of both `Vertical1` and `OrphanVert`
+        self.assertIn(multi_parent_html.location, orphan_vertical.children)
+        self.assertIn(multi_parent_html.location, vertical1.children)
+
+        # HTML component has `vertical1` as its parent.
+        html_parent = self.store.get_parent_location(multi_parent_html.location)
+        self.assertNotEqual(unicode(html_parent), unicode(orphan_vertical.location))
+        self.assertEqual(unicode(html_parent), unicode(vertical1.location))
+
+        # Get path of the `multi_parent_html` & Verify it should not be None
+        path = path_to_location(self.store, multi_parent_html.location)
+        self.assertIsNotNone(path)
+        self.assertEquals(path[0], course.id)
+
+    @ddt.data(ModuleStoreEnum.Type.split)
+    def test_path_to_location_for_orphan_chapter(self, module_store):
+        r"""
+        Make sure that path_to_location works with a component having multiple chapter parents,
+        from which one of them is orphan
+
+         course
+            |
+        chapter   chapter
+           |         |
+        vertical  vertical
+              \    /
+               html
+
+        """
+        # Get a course with orphan modules
+        course = self.create_course_with_orphans(module_store)
+        orphan_chapter = self.store.get_item(BlockUsageLocator(course.id, 'chapter', 'OrphanChapter'))
+        vertical1 = self.store.get_item(BlockUsageLocator(course.id, 'vertical', 'Vertical1'))
+
+        # Verify `OrhanChapter` is an orphan
+        self.assertIn(orphan_chapter.location, self.store.get_orphans(course.id))
+
+        # Create a vertical (`Vertical0`) in orphan chapter (`OrphanChapter`).
+        # OrphanChapter -> Vertical0
+        vertical0 = self.store.create_child(self.user.id, orphan_chapter.location, 'vertical', "Vertical0")
+        self.store.publish(vertical0.location, self.user.id)
+
+        # Create a component in `Vertical0`
+        # OrphanChapter -> Vertical0 -> Html
+        html = self.store.create_child(self.user.id, vertical0.location, 'html', "HTML0")
+        self.store.publish(html.location, self.user.id)
+
+        # Also make `Vertical1` the parent of `HTML0`. So `HTML0` will have to parents (`Vertical0` & `Vertical1`)
+        vertical1.children.append(html.location)
+        self.store.update_item(vertical1, self.user.id)
+
+        # Get parent location & verify its either of the two verticals. As both parents are non-orphan,
+        # alphabetically least is returned
+        html_parent = self.store.get_parent_location(html.location)
+        self.assertEquals(unicode(html_parent), unicode(vertical1.location))
+
+        path = path_to_location(self.store, html.location)
+        self.assertIsNotNone(path)
+        self.assertEquals(path[0], course.id)
```

Comment:
The reason I think the `len(path)` should not be `==1` because `path_to_location` method returns a tuple. 
> a tuple (course_id, chapter, section, position) is returned which is suitable for the courseware index view.

If at any point, `get_parent_location` find outs that there is an orphan in xblocks' parent/super parent it will return `None`. So its either a tuple or None which is returned.
[common/lib/xmodule/xmodule/modulestore/search.py#L113](https://github.com/edx/edx-platform/blob/756008aa9bd03ecd147ae98271a16fb640d59945/common/lib/xmodule/xmodule/modulestore/search.py#L113)